delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/client/DeltaClient.scala

```diff
@@ -1,44 +1,99 @@
 package ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.client
 
+import akka.actor.typed.ActorSystem
 import akka.http.scaladsl.client.RequestBuilding.Get
 import akka.http.scaladsl.model.ContentTypes.`application/json`
-import akka.http.scaladsl.model.HttpRequest
 import akka.http.scaladsl.model.headers.Accept
-import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.client.DeltaClient.accept
+import akka.http.scaladsl.model.{HttpRequest, HttpResponse}
+import akka.persistence.query.{NoOffset, Offset, Sequence, TimeBasedUUID}
+import akka.stream.alpakka.sse.scaladsl.EventSource
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.client.DeltaClient._
 import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.model.CompositeViewSource.RemoteProjectSource
 import ch.epfl.bluebrain.nexus.delta.rdf.RdfMediaTypes
-import ch.epfl.bluebrain.nexus.delta.sdk.http.{HttpClient, HttpClientError}
+import ch.epfl.bluebrain.nexus.delta.sdk.http.HttpClient
+import ch.epfl.bluebrain.nexus.delta.sdk.http.HttpClient.HttpResult
 import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.AuthToken
 import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectCountsCollection.ProjectCount
 import ch.epfl.bluebrain.nexus.delta.sdk.syntax._
-import monix.bio.IO
+import com.typesafe.scalalogging.Logger
+import io.circe.Decoder
+import io.circe.parser.decode
+import fs2._
+import monix.bio.{IO, Task}
+import monix.execution.Scheduler
+import streamz.converter._
+
+import java.util.UUID
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+import scala.util.Try
 
 /**
   * A client to request a remote delta instance
   */
-final class DeltaClient(client: HttpClient) {
+final class DeltaClient(client: HttpClient, retryDelay: FiniteDuration)(implicit
+    as: ActorSystem[Nothing],
+    scheduler: Scheduler
+) {
 
   /**
     * Fetches statistics for the remote source
     */
-  def statistics(source: RemoteProjectSource): IO[HttpClientError, ProjectCount] = {
-    implicit val cred: Option[AuthToken] = source.token.map { token => AuthToken(token.value.value) }
+  def statistics(source: RemoteProjectSource): HttpResult[ProjectCount] = {
+    implicit val cred: Option[AuthToken] = token(source)
     val statisticsEndpoint: HttpRequest  =
       Get(
         source.endpoint / "projects" / source.project.organization.value / source.project.project.value / "statistics"
       ).addHeader(accept).withCredentials
     client.fromJsonTo[ProjectCount](statisticsEndpoint)
   }
 
+  private def toOffset(id: String): Offset =
+    Try(TimeBasedUUID(UUID.fromString(id))).orElse(Try(Sequence(id.toLong))).getOrElse(NoOffset)
+
+  def events[A: Decoder](source: RemoteProjectSource, offset: Offset): Stream[Task, (Offset, A)] = {
```

Comment:
What is this `A` in our case. It is `A <: Event` ? I don't think `Decoder` is going to be enough here, since we cannot produce a `Decoder` for subtypes of Event. That's why we have `EventExchange`.
What we need the `events` method to return is a `Envelope[Event]`.
 
We will also need something like:
```scala
def fetchResource(id: Iri, project: ProjectRef, tag: Option[TagLabel]): Option[EventExchangeValue[_, _]] = ???
```

Notice that in the very end, what we need for the composite views is to be able to produce an `IndexingSource` from a `RemoteProjectSource` and an offset.

Reply:
Basically we need this same logic [IndexingStream](https://github.com/BlueBrain/nexus/blob/master/delta/sdk-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/views/indexing/IndexingSource.scala#L57) but with the following differences:
- The stream is not coming from a DB query, but from an HTTP query.
- The `toResource` method has to make an HTTP query to fetch the resource, and after that use the `EventExchange` for the conversion

Reply:
It can't really return an `Envelope[Event]` as SSE's don't provide akka-persistence fields.
To decode in this context, we don't need to deserialize the whole event, we just need to extract the id right ? We don't care about other fields, we just want to be able to make a second call to the remote delta instance to get the whole resource.

Reply:
We need the following:
- offset
- instant
- persistenceId
- sequenceNr
- id
- projectRef

If  we know what we want to return, we don't need an `A`, but a specific case class.

Reply:
We really have to go though this Envelope/EventExchange thing for a remote project source ?
When fetching the resource, we can fetch the expanded form and from this, get the graph and whatever we need to put it in blazegraph ?

Reply:
It is not as simple as that, since we need metadata and not metadata, we need some fields to build the correct `Message` and not to say that all logic is well tight together with the `EventExchange` currently.